charms/worker/k8s/src/token_distributor.py

```diff
@@ -245,12 +269,20 @@ def allocate_tokens(
                     relation.name,
                     unit.name,
                 )
-                app_databag[unit.name] = name
-                if invalidate_on_join:
-                    relation.data[self.charm.unit].pop(secret_id, None)
+                app_databag[unit.name] = f"joined-{name}"
+                if revoke_on_join:
+                    self._revoke_juju_secret(relation, unit)
+
```

Comment:
The remote unit is joined.  That means that token was used.  Let's CLEAN up the secret